Sources/WalletConnectAuth/Engine/Common/PairingEngine.swift

```diff
@@ -209,6 +212,7 @@ final class PairingEngine {
                 try? pairing.updateExpiry()
             }
             
+            onApprovalResponse?(proposal) // FIXME: This proposal needs to be stored on creation and retrieved from cache here. Don't use proposal from response.
```

Comment:
`This proposal needs to be stored on creation and retrieved from cache here. Don't use proposal from response.`
actually that proposal is from cache(jsonRpcHistory record) not from response, response do not contain proposal.